packages/beacon-node/src/chain/prepareNextSlot.ts

```diff
@@ -257,9 +257,9 @@ export class PrepareNextSlotScheduler {
    * Schedule task to update payload with inclusion list transactions that gathered up to `PROPOSER_INCLUSION_LIST_CUT_OFF`
    */
   async schedulePayloadInclusionListUpdate(payloadId: PayloadId, clockSlot: Slot): Promise<void> {
-    // TODO EIP-7805: Hard-coding sleepTime to 500ms for testing purpose. Revert it back after done testing
-    // const sleepTime = Math.max(0.5, (this.config.PROPOSER_INCLUSION_LIST_CUT_OFF - this.chain.clock.secFromSlot(clockSlot)) * 1000);
-    const sleepTime = 0.5 * 1000;
+    const secToNextSlot = this.config.SECONDS_PER_SLOT - this.chain.clock.secFromSlot(clockSlot);
+    const secToCutOff = this.config.PROPOSER_INCLUSION_LIST_CUT_OFF - this.chain.clock.secFromSlot(clockSlot);
+    const sleepTime = Math.min(secToNextSlot, secToCutOff) * 1000;
     await sleep(sleepTime, this.signal);
```

Comment:
why can't we just do what I did previously? Like we either wait until reach the cutoff or otherwise we just don't sleep at all if for some reason we are past the cut off already (which shouldn't happen)

```suggestion
    await sleep(
      Math.max(0, (this.config.PROPOSER_INCLUSION_LIST_CUT_OFF - this.chain.clock.secFromSlot(clockSlot)) * 1000),
      this.signal
    );
```

Reply:
> why can't we just do what I did previously? We either wait until we reach the cutoff or otherwise we just don't sleep at all if for some reason we are past the cut off already (which shouldn't happen)

Because the kurtosis config we use when interopping/testing use mainnet preset with `seconds_per_slot = 6`.

So `prepareNextSlot` usually happens at 4s, and `this.config.PROPOSER_INCLUSION_LIST_CUT_OFF - this.chain.clock.secFromSlot(clockSlot)` usually yield `11 - 2 = 9` seconds of wait time. This would delay this EL call to 2 slots after.

Reply:
we should probably change this already on the spec, `PROPOSER_INCLUSION_LIST_CUT_OFF` and other timing parameters should not be part of config and always derived from `SECONDS_PER_SLOT` similar to proposer boost reorg cutoff
https://github.com/ChainSafe/lodestar/blob/8e915fe6fcd77d4e92a4905e57e0cb5fdff99f20/packages/fork-choice/src/forkChoice/forkChoice.ts#L311

Reply:
resolved by https://github.com/ChainSafe/lodestar/pull/8478